lib/services/access_list.go

```diff
@@ -218,47 +233,206 @@ func UnmarshalAccessListMember(data []byte, opts ...MarshalOption) (*accesslist.
 	return &member, nil
 }
 
-// IsAccessListOwner will return true if the user is an owner for the current list.
-func IsAccessListOwner(identity tlsca.Identity, accessList *accesslist.AccessList) error {
+// IsAccessListOwner will determine if the user is an owner for the current list.
+// If ownership is inherited (through the user having Membership of a list which is an Owner of the provided `accessList`),
+// the user must also meet the ownership requirements of the parent list.
+func IsAccessListOwner(ctx context.Context, members AccessListsAndMembersGetter, identity tlsca.Identity, accessList *accesslist.AccessList) (bool, error) {
```

Comment:
This function seems to be broken. 

if a user is top lever owner 

the following isOwner var will be set to true: 
```
isOwner := slices.ContainsFunc(accessList.GetOwners(), func(owner accesslist.Owner)
```

then we have following logic:
```
inherited := false
if !isOwner {
  inherited = true
}
return inherited
```

In the result in isOwner is true the inherited == false arg is returned from `IsAccessListOwner`